tutorials/misc/plot_ecog.py

```diff
@@ -9,71 +11,152 @@
+# First, use pandas to read in the .tsv file.
 
-mat = loadmat(mne.datasets.misc.data_path() + '/ecog/sample_ecog.mat')
-ch_names = mat['ch_names'].tolist()
-elec = mat['elec']  # electrode positions given in meters
-# Now we make a montage stating that the sEEG contacts are in head
-# coordinate system (although they are in MRI). This is compensated
-# by the fact that below we do not specicty a trans file so the Head<->MRI
-# transform is the identity.
-montage = mne.channels.make_dig_montage(ch_pos=dict(zip(ch_names, elec)),
-                                        coord_frame='head')
+# In this tutorial, the electrode coordinates are assumed to be in meters
+elec_df = pd.read_csv(misc_path + '/ecog/sample_ecog_electrodes.tsv',
+                      sep='\t', header=0, index_col=None)
+ch_names = elec_df['name'].tolist()
+ch_coords = elec_df[['x', 'y', 'z']].to_numpy(dtype=float)
+ch_pos = dict(zip(ch_names, ch_coords))
+
+###############################################################################
+# Now we make a :class:`mne.channels.DigMontage` stating that the ECoG
+# contacts are in head coordinate system (although they are in MRI). This is
+# compensated below by the fact that we do not specify a trans file so the
+# Head<->MRI transform is the identity.
+
+montage = mne.channels.make_dig_montage(ch_pos, coord_frame='head')
 print('Created %s channel positions' % len(ch_names))
 
 ###############################################################################
-# Now that we have our electrode positions in MRI coordinates, we can create
-# our measurement info structure.
+# Now that we have our montage, we can load in our corresponding
+# time-series data and set the montage to the raw data.
+
+# first we'll load in the sample dataset
+raw = mne.io.read_raw_edf(misc_path + '/ecog/sample_ecog.edf')
+
+# drop bad channels
+raw.info['bads'].extend([ch for ch in raw.ch_names if ch not in ch_names])
+raw.load_data()
+raw.drop_channels(raw.info['bads'])
+raw.crop(0, 2)  # just process 2 sec of data for speed
 
-info = mne.create_info(ch_names, 1000., 'ecog').set_montage(montage)
+# attach montage
+raw.set_montage(montage)
 
 ###############################################################################
 # We can then plot the locations of our electrodes on our subject's brain.
 #
 # .. note:: These are not real electrodes for this subject, so they
 #           do not align to the cortical surface perfectly.
+#
+# Sometimes it is useful to make a scatterplot for the current figure view.
+# This is best accomplished with matplotlib. We can capture an image of the
+# current mayavi view, along with the xy position of each electrode, with the
+# `snapshot_brain_montage` function. We can then visualize for example the
+# gamma power on the brain using MNE and matplotlib functions.
+# Here, we'll use :func:`~mne.viz.snapshot_brain_montage` to save the plot
+# as image data, so that later in the tutorial we can plot data on top of it.
```

Comment:
This reads awkwardly. Suggestion:

```suggestion
###############################################################################
# We can then plot the locations of our electrodes on our subject's brain.
# We'll use :func:`~mne.viz.snapshot_brain_montage` to save the plot as image
# data (along with xy positions of each electrode in the image), so that later
# we can plot frequency band power on top of it.
#
# .. note:: These are not real electrodes for this subject, so they
#           do not align to the cortical surface perfectly.
```